src/utils/communitySbt/getTopBadges.ts

```diff
@@ -0,0 +1,201 @@
+                        termEndTimestamp
+                        rateOracle {
+                            token {
+                                name
+                                decimals
+                            }
+                        }
+                    }
+                    mints {
+                        transaction {
+                            createdTimestamp
+                        }
+                        amount
+                    }
+                    burns {
+                        transaction {
+                            createdTimestamp
+                        }
+                        amount
+                    }
+                    swaps {
+                        transaction {
+                            createdTimestamp
+                        }
+                        cumulativeFeeIncurred
+                        variableTokenDelta
+                    }
+                }
+            }
+        }
+      `
+
+    const client = new ApolloClient({
+        cache: new InMemoryCache(),
+        link: new HttpLink({ uri: subgraphUrl, fetch })
+    })
+  
+    const scores: Record<string, number> = {};
+  
+    let skip = 0;
+    while (true) {
+        const data = await client.query({
+            query: gql(activityQuery),
+            variables: {
+                skipCount: skip,
+            },
+        });
+      skip += 1000;
+  
+      for (const wallet of  data.data.wallets) {
+        let score = 0;
+  
+        for (const position of wallet.positions) {
+          const token = position.amm.rateOracle.token.name;
+          const decimals: number = position.amm.rateOracle.token.decimals;
+  
+          const termEnd = Number(
+            ethers.utils.formatUnits(position.amm.termEndTimestamp.toString(), 18),
+          );
+
+          let args : UpdateScoreArgs = {
+            score: score,
+            actions: position.burns as MintOrBurnAction[],
+            actionType: ActionType.BURN,
+            decimals: decimals,
+            token: token,
+            seasonStart: seasonStart,
+            seasonEnd: seasonEnd,
+            termEnd: termEnd,
+            ethPrice: ethPrice
+          }
+          if (isLP) {
+            const burnArgs : UpdateScoreArgs  = {...args, actionType: ActionType.BURN, actions:  position.burns as MintOrBurnAction[]};
+            score = updateScore(burnArgs);
+            const mintArgs : UpdateScoreArgs  = {...args, actionType: ActionType.MINT, actions:  position.mints as MintOrBurnAction[]};
+            score = updateScore(mintArgs);
+          } else {
+            const swapArgs : UpdateScoreArgs  = {...args, actionType: ActionType.SWAP, actions:  position.swaps as SwapAction[]};
+            score = updateScore(swapArgs);
+          }
+        }
+  
+        if (score > 0 && !ignoredWalletIds[wallet.id.toLowerCase()]) {
+            scores[wallet.id] = score;
+        }
+      }
+  
+      if (data.data.wallets.length < 1000) {
```

Comment:
what if data.data.wallets.length breaks due to data being null, data.data being null, data.data.wallets being null. I would use ? operator here

Reply:
true, I added a check on this: if null -> empty array (c856fd63bdec36d55495e3009171c2af0029d365)